baseapp/options.go

```diff
@@ -406,3 +406,14 @@ func (app *BaseApp) SetMsgServiceRouter(msgServiceRouter *MsgServiceRouter) {
 func (app *BaseApp) SetGRPCQueryRouter(grpcQueryRouter *GRPCQueryRouter) {
 	app.grpcQueryRouter = grpcQueryRouter
 }
+
+func (app *BaseApp) SetPrivValidatorKeyGen(keyGenF KeyGenF) {
```

Comment:
not sure the best approach here as its limited key types, not everything is supported and it requires consensus params changes.